### ISSUE TITLE
Sending custom event support

### ISSUE THREAD

**Issue by evan-waters** [seegno/primus-emitter-objc#1]
The example in the `README` shows listening to custom events, but there is no mention of sending custom events. Is this supported or will I need manually to wrap messages to be sent to _primus-emitter_ on the server?

**Comment by nunofgs** [comment]
Hi @evro.

You are absolutely correct. This library only supports listening to custom events currently. We did not have the need to implement the _send_ method, so we ended up putting it off.

However, it shouldn't be too hard to implement it (basically, porting the `.send()` [method](https://github.com/cayasso/primus-emitter/blob/master/lib/index.js#L41)).

If you'd be willing to implement it, please give it a shot and submit a PR. I'll help in any way I can.

**Comment by danil-z** [comment]
Give me a hand. Could you implement a simple example 
spark.send('ping', data);

**Comment by nunofgs** [comment]
@blackbumer: Sorry for taking a couple of days to reply. I've [submitted a PR](https://github.com/seegno/primus-emitter-objc/pull/2) with initial support for the _send_ method.

@evro, @blackbumer Can you test the PR and report back?

**Comment by naartjie** [comment]
Ah, just what I've been looking for ;-) I'll give it a whirl and report back as well.

**Comment by naartjie** [comment]
Hmmm, doesn't work for me. I'm using primus `v2.4.12` and primus-emitter `v3.0.3` on the server.

When I try:

``` js
spark.on('event-name', function(data) {
  console.log(data)
}
```

I get nothing back.

However:

``` js
spark.on('incoming::data', function(data) {
  console.log(data.toString())
}
```

yields: `{"type":0,"data":["event-name","data string"]}`

@nunofgs Am I missing something obvious?

**Comment by naartjie** [comment]
Hola, it's me again...

I figured out why it wasn't working... the data was arriving as a `Buffer` on the server. For me the workaround was to change `raw` to `NSString` instead of `NSData` before being returned, so I added this line after [`JSON.m:21`](https://github.com/seegno/primus-objc/blob/2d625a5cdf0f69fc8ac11576ca098e97a4fbb41c/Primus/Parsers/JSON/JSON.m#L21):

``` Objective-C
raw = [[NSString alloc] initWithData:raw encoding:NSUTF8StringEncoding];
```

I will make a PR for this to be clear,  but I don't know how this will behave in other scenarios.

**Comment by naartjie** [comment]
PR submitted: seegno/primus-objc#8

**Comment by nunofgs** [comment]
I've just released version _primus-emitter-objc_ 0.0.5 and _primus-emitter-objc_ 0.0.7 which includes support for sending custom events.

Thanks guys!

**Comment by naartjie** [comment]
Nice, thanks!! :+1:
